pysteps/blending/steps.py

```diff
@@ -1440,19 +1440,27 @@ def calculate_weights_spn(correlations, cov):
             raise ValueError("cov must contain a covariance matrix")
         else:
             # Make a numpy matrix out of cov and get the inverse
+            cov = np.where(cov == 0.0, 10e-5, cov)
+            if cov[0][0] != 1.0:
+                cov[0][0] = 1.0
+            if cov[1][1] != 1.0:
+                cov[1][1] = 1.0
```

Comment:
Can you briefly explain lines 1444-1447? Beside it, I guess the if statements are not really necessary? Just set the values to 1, no?

Reply:
Just setting it to 1.0, works as well! I noticed that in some cases, with e.g. hardly any rain in the NWP forecast, np.corrcoef does not return a correlation of 1.0 for the model with itself. This is to ensure that is the case.